Kinvey.Tests/DataStoreTests/DataStoreNetworkIntegrationTests.cs

```diff
@@ -5089,204 +5234,247 @@ public async Task TestNetworkStoreGetSumAsync()
             Assert.AreEqual(1, arrGAR.Count());
         }
 
-		[TestMethod]
+        #region Save
+
+        [TestMethod]
 		public async Task TestSaveAsync()
 		{
             // Setup
+            kinveyClient = BuildClient();
+
             if (MockData)
             {
                 MockResponses(3);
             }
 			await User.LoginAsync(TestSetup.user, TestSetup.pass, kinveyClient);
 
-			// Arrange
-			ToDo newItem = new ToDo();
-			newItem.Name = "Next Task";
-			newItem.Details = "A test";
-			newItem.DueDate = "2016-04-19T20:02:17.635Z";
-			DataStore<ToDo> todoStore = DataStore<ToDo>.Collection(collectionName, DataStoreType.NETWORK);
+            // Arrange
+            var newItem = new ToDo
+            {
+                Name = "Next Task",
+                Details = "A test",
+                DueDate = "2016-04-19T20:02:17.635Z"
+            };
+            var todoStore = DataStore<ToDo>.Collection(collectionName, DataStoreType.NETWORK);
 
 			// Act
-			ToDo savedToDo = await todoStore.SaveAsync(newItem);
-
-			// Assert
-			Assert.IsNotNull(savedToDo);
-			Assert.IsTrue(string.Equals(savedToDo.Name, newItem.Name));
-
+			var savedToDo = await todoStore.SaveAsync(newItem);
+			
 			// Teardown
 			await todoStore.RemoveAsync(savedToDo.ID);
-			kinveyClient.ActiveUser.Logout();
-		}
 
-		[TestMethod]
-		public async Task TestStoreInvalidOperation()
-		{
+            // Assert
+            Assert.IsNotNull(savedToDo);
+            Assert.AreEqual(savedToDo.Name, newItem.Name);
+        }
+
+        [TestMethod]
+        public async Task TestSaveCreateApiVersion4Async()
+        {
             // Setup
+            kinveyClient = BuildClient("4");
+
             if (MockData)
             {
-                MockResponses(1);
+                MockResponses(4);
             }
-			await User.LoginAsync(TestSetup.user, TestSetup.pass, kinveyClient);
 
-			DataStore<ToDo> todoStore = DataStore<ToDo>.Collection(collectionName, DataStoreType.NETWORK, kinveyClient);
+            await User.LoginAsync(TestSetup.user, TestSetup.pass, kinveyClient);
 
-            await Assert.ThrowsExceptionAsync<KinveyException>(async delegate ()
-			{
-				await todoStore.PullAsync();
-			});
+            // Arrange
+            var newItem = new ToDo
+            {
+                Name = "Next Task",
+                Details = "A test",
+                DueDate = "2016-04-19T20:02:17.635Z"
+            };
+            var todoStore = DataStore<ToDo>.Collection(collectionName, DataStoreType.NETWORK);
 
-            await Assert.ThrowsExceptionAsync<KinveyException>(async delegate ()
-			{
-				await todoStore.PushAsync();
-			});
+            // Act
+            var savedToDo = await todoStore.SaveAsync(newItem);
 
-            await Assert.ThrowsExceptionAsync<KinveyException>(async delegate ()
-			{
-				await todoStore.SyncAsync();
-			});
-		}
+            var existingToDo = await todoStore.FindByIDAsync(savedToDo.ID);
```

Comment:
I think it would be better to use just Find and assert the number of items - if for some reason two item is recorded multiple times

Reply:
Ok.